pkg/scheduler/objects/application.go

```diff
@@ -2083,3 +2087,40 @@ func getRateLimitedAppLog() *log.RateLimitedLogger {
 	})
 	return rateLimitedAppLog
 }
+
+func (sa *Application) updateRunnableStatus(runnableInQueue, runnableByUserLimit bool) {
+	sa.Lock()
+	defer sa.Unlock()
+	if sa.runnableInQueue != runnableInQueue {
+		sa.runnableInQueue = runnableInQueue
+		if sa.runnableInQueue {
```

Comment:
we can use `runnableInQueue` to replace `sa.runnableInQueue` here, and then we can remove line #2095 to avoid duplicate assignment (`sa.runnableInQueue = runnableInQueue`)

Reply:
for example:
```go
func (sa *Application) updateRunnableStatus(runnableInQueue, runnableByUserLimit bool) {
	sa.Lock()
	defer sa.Unlock()
	if sa.runnableInQueue != runnableInQueue {
		if runnableInQueue {
			log.Log(log.SchedApplication).Info("Application is now runnable in queue",
				zap.String("appID", sa.ApplicationID),
				zap.String("queue", sa.queuePath))
			sa.appEvents.sendAppRunnableInQueueEvent()
		} else {
			log.Log(log.SchedApplication).Info("Maximum number of running applications reached the queue limit",
				zap.String("appID", sa.ApplicationID),
				zap.String("queue", sa.queuePath))
			sa.appEvents.sendAppNotRunnableInQueueEvent()
		}
	}

	if sa.runnableByUserLimit != runnableByUserLimit {
		if runnableByUserLimit {
			log.Log(log.SchedApplication).Info("Application is now runnable based on user quota",
				zap.String("appID", sa.ApplicationID),
				zap.String("queue", sa.queuePath),
				zap.String("user", sa.user.User),
				zap.Strings("groups", sa.user.Groups))
			sa.appEvents.sendAppRunnableQuotaEvent()
		} else {
			log.Log(log.SchedApplication).Info("Maximum number of running applications reached the user/group limit",
				zap.String("appID", sa.ApplicationID),
				zap.String("queue", sa.queuePath))
			sa.appEvents.sendAppNotRunnableQuotaEvent()
		}
	}
	sa.runnableInQueue = runnableInQueue
	sa.runnableByUserLimit = runnableByUserLimit
}
```